db/revision_cache.go

```diff
@@ -68,21 +84,37 @@ func (rc *RevisionCache) Get(docid, revid string) (DocumentRevision, error) {
 	if value == nil {
 		return DocumentRevision{}, nil
 	}
-	docRev, err := value.load(rc.loaderFunc)
+	docRev, err := value.load(rc.loaderFunc, rc.statsRecorderFunc)
+
 	if err != nil {
 		rc.removeValue(value) // don't keep failed loads in the cache
 	}
 	return docRev, err
 }
 
+func (rc *RevisionCache) statsRecorderFunc(hitOrMiss RevCacheStatsEvent) {
```

Comment:
I think the callback function/statsEvent model is overkill and inefficient.  We're really just incrementing two counters (hit, miss).  Either rc.statsCache to value.load, or have value.load return a boolean for hit/miss.

Reply:
Agreed it's a bit clunky, but it seemed like a better alternative than passing in an `rc *RevisionCache` as a param which makes this method harder to unit test.  

How about converting from a function callback to an interface?

```
func (value *revCacheValue) load(loaderFunc RevisionCacheLoaderFunc, statsRecorder StatsRecorder) (DocumentRevision, error) {
```

Where  `StatsRecorder` is defined as:

```
type StatsRecorder interface {
  CacheHit()
  CacheMiss()
}
```

and `rc *RevisionCache` would implement those methods be storing into it's internal `statsCache *expvar.Map`?

Reply:
Switched over to hit/miss returned as a bool by the load functions.  Sorry, I'd missed that in the first reading.